File: app/src/main/java/com/kupay/kupay/common/js/CallJSRunnable.java
package com.kupay.kupay.common.js;

/**
 * Created by "iqos_jay@outlook.com" on 2018/6/27.
 */
public class CallJSRunnable implements Runnable {

    private final String func;

    public CallJSRunnable(String func) {
        this.func = func;
    }

    @Override
    public void run() {
        try {
            android.webkit.WebView webView = (android.webkit.WebView) JSEnv.getEnv(JSEnv.WEBVIEW);
            webView.evaluateJavascript(func, null);
        } catch (Exception e) {
            com.tencent.smtt.sdk.WebView webView = (com.tencent.smtt.sdk.WebView) JSEnv.getEnv(JSEnv.WEBVIEW);
            webView.evaluateJavascript(func, null);
            e.printStackTrace();
        }
    }
}

File: app/src/main/java/com/kupay/kupay/common/js/CallJSRunnable.kt
package com.kupay.kupay.common.js

import com.kupay.kupay.base.BaseWebView
import com.kupay.kupay.widget.AndroidWebView
import com.kupay.kupay.widget.X5Chrome

/**
 * Created by "iqos_jay@outlook.com" on 2018/6/27.
 */
class CallJSRunnable(private val func: String) : Runnable {

    override fun run() {
        if (BaseWebView.isX5) {
            val webView = JSEnv.getEnv(JSEnv.WEBVIEW) as X5Chrome
            webView.evaluateJavascript(func, null)
        } else {
            val webView = JSEnv.getEnv(JSEnv.WEBVIEW) as AndroidWebView
            webView.evaluateJavascript(func, null)
        }
    }
}
